library/src/engine.cxx

```diff
@@ -39,18 +39,29 @@ engine::engine(
   // Ensure all lib initialization is done (once)
   detail::init::initialize();
 
-  // build default cache path
 #if defined(_WIN32)
-  std::string cachePath = vtksys::SystemTools::GetEnv("LOCALAPPDATA");
-  cachePath = cachePath + "/f3d";
+  static constexpr const char* CACHE_ENV_VAR = "LOCALAPPDATA";
+#else
+  static constexpr const char* CACHE_ENV_VAR = "HOME";
+#endif
+
+  char* env = std::getenv(CACHE_ENV_VAR);
+  if (!env)
+  {
+    throw engine::cache_exception(
+      std::string("Could not setup cache, please set ") + CACHE_ENV_VAR + " environment variable");
+  }
+
+  fs::path cachePath(env);
+
+#if defined(_WIN32)
+  cachePath /= "f3d";
 #elif defined(__APPLE__)
-  std::string cachePath = vtksys::SystemTools::GetEnv("HOME");
-  cachePath += "/Library/Caches/f3d";
+  cachePath = cachePath / "Library" / "Caches" / "f3d";
 #elif defined(__ANDROID__)
-  std::string cachePath = ""; // no default
+//  fs::path cachePath;
```

Comment:
no default does not makes much sense, we cannot force libf3d user to set a cachePath. Is there no reasonnable cache path we could use by default ?

Reply:
@Meakk